file/hasher/reference.go

```diff
@@ -78,22 +78,54 @@ func (r *ReferenceHasher) sum(lvl int) []byte {
 // skips intermediate levels that end on span boundary
 func (r *ReferenceHasher) digest() []byte {
 
-	// if we did not end on a chunk boundary, the last chunk hasn't been hashed
-	// we need to do this first
+	// if we didn't end on a chunk boundary we need to hash remaining chunks first
+	r.hashUnfinished()
```

Comment:
this should probably be called before `r.digest()` no?

Reply:
yeah %) @acud now?

Reply:
you're a sweetheart @nolash